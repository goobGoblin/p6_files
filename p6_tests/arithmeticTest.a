arithmeticTest : () -> int {
  a : int;
  b : int;
  c : int;
  d : int;
  e : int;
  f : int;
  
  a = 10;
  b = 5;
  
  c = -a;
  
  d = a + b;
  e = a - b;
  f = a * b;
  a = a / b;
  
  b = (c + d) * (e - f) / a;
  
  return b;
}
