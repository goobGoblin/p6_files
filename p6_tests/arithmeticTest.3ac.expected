[BEGIN GLOBALS]
arithmeticTest
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
lbl_0:      leave <init>
[BEGIN arithmeticTest LOCALS]
a (local var of 8 bytes)
b (local var of 8 bytes)
c (local var of 8 bytes)
d (local var of 8 bytes)
e (local var of 8 bytes)
f (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
tmp2 (tmp var of 8 bytes)
tmp3 (tmp var of 8 bytes)
tmp4 (tmp var of 8 bytes)
tmp5 (tmp var of 8 bytes)
tmp6 (tmp var of 8 bytes)
tmp7 (tmp var of 8 bytes)
tmp8 (tmp var of 8 bytes)
[END arithmeticTest LOCALS]
fun_arithmeticTest: enter arithmeticTest
            [a] := 10
            [b] := 5
            [tmp0] := NEG64 [a]
            [c] := [tmp0]
            [tmp1] := [a] ADD64 [b]
            [d] := [tmp1]
            [tmp2] := [a] SUB64 [b]
            [e] := [tmp2]
            [tmp3] := [a] MULT64 [b]
            [f] := [tmp3]
            [tmp4] := [a] DIV64 [b]
            [a] := [tmp4]
            [tmp5] := [c] ADD64 [d]
            [tmp6] := [e] SUB64 [f]
            [tmp7] := [tmp5] MULT64 [tmp6]
            [tmp8] := [tmp7] DIV64 [a]
            [b] := [tmp8]
            setret [b]
            goto lbl_1
lbl_1:      leave arithmeticTest
